HIP/hip-1056.md

```diff
@@ -0,0 +1,1727 @@
+
+## Motivation
+
+Block streams resolve many issues with the previous Record Stream v6 specification from [HIP 435](https://hips.hedera.com/hip/hip-435)
+currently deployed on mainnet. Below are some key issues along with the potential resolutions that block streams
+provide:
+
+- **Mixed Serialization Formats**: The record stream mixes protobuf with other serialization formats instead of using
+  pure protobuf serialization. Block streams will use only protobuf definitions, making it much easier to consume and
+  validate.
+- **Data Duplication**: Transaction data was redundantly duplicated between event streams and record streams, resulting
+  in wasted space and bandwidth. Block streams aim to reduce this duplication by being more explicit about the data
+  being serialized, and streamlining the format.
+- **Event Stream Documentation**: Currently, the Event stream is not in a publicly documented protobuf format. Block
+  streams integrate Event Stream data in protobuf which increases the transparency of consensus logic. Event streams
+  data can be used to validate how consensus was reached and that it was fair. They can also be used for event replay in
+  crash recovery and node restart operations.
+- **Signature Files**: Each node currently produces a v6 record stream with a corresponding signature file for
+  verification. Mirror nodes must download a strong minority (1/3 of consensus weight) worth of these signature files,
+  along with the record file, to verify v6 records. This frequent access to cloud storage (GCS/S3) is costly. To reduce
+  this expense and complexity, blocks include a TSS signature from the majority weight of consensus nodes.
+- **File Naming and Retrieval**: v6 record stream files are named after the consensus timestamp of the first transaction
+  they contain, requiring Mirror Nodes to perform frequent and costly LIST operations on cloud storage (GCS/S3) to find
+  new files. The new block streams use sequential block numbers, eliminating the need for these expensive LIST
+  operations. Mirror Nodes will read the stream from a block node that manages its own storage at a lower cost and won’t
+  need to deal with cloud storage buckets.
+- **Sidecar Data Integration**: In v6 record streams, sidecar files are created to serialize extra data such as smart
+  contract logs outside the main record stream. Block streams will incorporate sidecar data into the unified stream.
+- **Subset Support**: Any block node operator will be able to decide individually which data in the block stream to
+  retain on disk, without breaking the cryptographic integrity of the block stream. Different block node operators in
+  different legal jurisdictions can make different decisions about what data to retain. Or, block node operators may
+  subset the stream to retain minimal data and minimize storage costs.
+- **Errata Handling**: Errata refers to software bugs where a node executed the transaction correctly but did not record
```

Comment:
Alternative option would be the given block node to be temporarily excluded from the network of block nodes and it's block stream invalidated. Then, upon fixing the bug, the block node operator could again join the network, sync the state and continue it's normal work.

Reply:
Errata are needed when a bug on the _consensus node_ produces incorrect output. This section refers to the necessity to change the recorded block chain.  I believe this has been necessary a small number of times since genesis on mainnet.

Block Nodes are not a network; each node is completely independent.  Those details will be specified in a different HIP.

Reply:
@IvanKavaldzhiev will call out the non network block node in https://github.com/hashgraph/hedera-improvement-proposal/pull/1081
Feel free to make comments there also